### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-09-01

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#17]
# iPAS AI 應用規劃師科目3練習 - 2026-09-01（第 7 輪）

本次共 **10 題**，聚焦「L233 建模與參數調校」與「L234 機器學習治理」，並以「物流與交通」「公部門與智慧城市」兩種產業情境包裝。

**作答方式**：請直接點選每個選項前的核取方塊（`- [ ]` → `- [x]`）即可作答，**不需留言、不需進入編輯畫面、不需另填答案欄位**。單選題只勾 1 項；多選題請依題目標示的項數勾選。

**批改方式**：全部勾選完成後，請將本 issue 的 label 改為 `ai-analysis`，系統即會進行批改與錯題分析。

## 第 1 題〔單選〕考點：L23301 數據準備與特徵工程

某物流公司要用「配送區域（北區／中區／南區）」與「包裹類型（生鮮／一般／易碎）」等**無序類別特徵**，訓練一個以梯度下降最佳化的邏輯迴歸模型來預測是否延遲。下列編碼方式何者最適當？

- [ ] (A) 使用 Label Encoding 將類別映射為 0／1／2 等整數即可
- [ ] (B) 直接把原始中文字串丟入模型，交給模型自行處理
- [x] (C) 使用 One-Hot Encoding，為無序類別避免捏造大小順序
- [ ] (D) 只需對這些欄位做 StandardScaler 標準化即可

---

## 第 2 題〔單選〕考點：L23302 模型訓練與調校

智慧城市的交通流量預測模型在訓練集準確率 0.98、驗證集僅 0.71，且差距持續擴大。下列何者是最適當的改善方向？

- [ ] (A) 增加網路層數與神經元數量，以擬合更多細節
- [ ] (B) 移除驗證集，只依訓練集準確率判斷模型好壞
- [ ] (C) 持續延長訓練直到訓練準確率達到 100%
- [x] (D) 加入 L2 正則化與 Dropout，並蒐集更多訓練資料

---

## 第 3 題〔單選．程式／輸出判讀〕考點：L23303 模型評估指標

某物流平台以模型偵測「異常包裹」（1＝異常，0＝正常），執行下列程式：

```python
from sklearn.metrics import recall_score, precision_score

# 1 = 異常包裹, 0 = 正常包裹
y_true = [1, 1, 1, 1, 0, 0, 0, 0, 0, 0]
y_pred = [1, 1, 0, 0, 1, 0, 0, 0, 0, 0]

print(round(recall_score(y_true, y_pred), 2),
      round(precision_score(y_true, y_pred), 2))
```

請問這段程式的輸出最接近下列何者？

- [ ] (A) 0.5 0.67
- [x] (B) 0.67 0.5
- [ ] (C) 0.5 0.5
- [ ] (D) 0.2 0.2

---

## 第 4 題〔多選（可勾選 3 項）〕考點：L23304 模型佈署與監控

智慧城市的空氣品質預測模型上線半年後，離線與線上績效持續下滑，但程式碼與模型參數皆未更動，輸入資料的分布卻已改變。下列哪些是**正確**的處理方式？（請勾選 3 項）

- [x] (A) 建立輸入特徵分布與線上預測績效的持續監控
- [x] (B) 以近期資料重新訓練或更新模型
- [ ] (C) 判定為程式 bug，只要回滾到舊版程式碼即可解決
- [x] (D) 設定資料漂移／概念漂移的告警機制
- [ ] (E) 無需處理，模型績效一段時間後會自行回穩

---

## 第 5 題〔多選（可勾選 3 項）〕考點：L23301 數據準備與特徵工程

某物流公司在建模前評估「哪些演算法在訓練前通常需要先做特徵縮放（Normalization／Standardization）」。下列哪些演算法**對特徵尺度敏感、通常需要縮放**？（請勾選 3 項）

- [x] (A) KNN（K-最近鄰）
- [ ] (B) 決策樹（Decision Tree）
- [ ] (C) 以梯度下降訓練的邏輯迴歸（Logistic Regression）
- [x] (D) 隨機森林（Random Forest）
- [x] (E) K-Means

---

## 第 6 題〔單選〕考點：L23401 資料與模型治理

某市政府建置「智慧交通違規辨識系統」，需使用車牌影像與市民相關個資。下列何者最符合資料治理原則？

- [ ] (A) 盡量蒐集越多欄位越好，以提升模型準確率
- [x] (B) 遵循資料最小化原則，僅蒐集必要欄位並進行去識別化與權限控管
- [ ] (C) 將完整原始個資公開給所有市府部門自由取用
- [ ] (D) 為求作業便利，模型輸入資料不需任何存取權限控管

---

## 第 7 題〔情境分析．單選〕考點：L23402 AI 風險與倫理

某市政府導入 AI 自動核定社會福利補助。上線後發現：某弱勢族群的通過率明顯低於其他族群，系統無法提供決策說明，民眾也沒有申訴管道。下列處理方式何者最適當？

- [ ] (A) 直接採用系統結果，因為整體 Accuracy 很高
- [x] (B) 檢查訓練資料是否存在歷史／抽樣偏差，以公平性指標分群評估，並提供可解釋性、申訴機制與人為監督
- [ ] (C) 刪除該弱勢族群的測試資料，讓群體間差異在報表上消失
- [ ] (D) 只要增加模型層數與訓練時間，公平性問題即可自動解決

---

## 第 8 題〔多選（可勾選 4 項）〕考點：L23401 資料與模型治理

某物流平台要為其配送與客戶資料建立完整的資料治理機制。下列哪些屬於**負責任的資料治理**作法？（請勾選 4 項）

- [x] (A) 資料最小化，只保留模型所需的必要欄位
- [x] (B) 對個人資料進行去識別化／匿名化
- [x] (C) 建立資料存取權限控管與稽核紀錄
- [ ] (D) 將客戶完整住址與電話公開於內部共享區，供全體員工自由取用
- [x] (E) 建立資料與模型的血緣（lineage）追蹤與版本紀錄

---

## 第 9 題〔單選〕考點：L23202 常見機器學習演算法

某物流公司需要一個模型判斷訂單風險，並要能**向稽核人員清楚說明「為何某筆訂單被判為高風險」的決策規則**（以條件式判斷呈現）。在下列選項中，最符合此可解釋性需求的是？

- [ ] (A) 深度神經網路（Deep Neural Network）
- [x] (B) SVM（RBF 核）
- [ ] (C) 決策樹（Decision Tree）
- [ ] (D) KNN（K-最近鄰）

---

## 第 10 題〔單選〕考點：L23101 機率／統計（貝氏定理・基率謬誤）

智慧城市的路口違規偵測系統中，真實違規率（盛行率）為 5%；當車輛確實違規時，系統偵測到違規的機率為 80%（敏感度）；當車輛未違規時，系統誤報為違規的機率為 10%。若系統對某車輛發出違規警報，該車輛**實際違規**的機率最接近下列何者？

- [ ] (A) 約 80%
- [x] (B) 約 55%
- [ ] (C) 約 30%
- [ ] (D) 約 5%

**Comment by JohnnyLai92** [comment]
**Claude Code 練習批改與題型分析**

已完成批改。以下為評語內容（不修改任何檔案）：

---

### 批改結果

| 題號 | 你的作答 | 正確答案 | 判定 | 解析 |
|---|---|---|---|---|
| 1 | C | **C** | ✅ | 無序類別應用 One-Hot，避免 Label Encoding 對梯度模型捏造大小順序。 |
| 2 | D | **D** | ✅ | 訓練高、驗證低且差距擴大＝過擬合，正解為正則化（L2/Dropout）＋增加資料。 |
| 3 | B | **A** | ❌ | 見下方詳解。 |
| 4 | A,B,D | **A,B,D** | ✅ | 資料漂移情境：持續監控、重訓、設漂移告警；回滾程式碼與「自行回穩」皆錯。 |
| 5 | A,D,E | **A,C,E** | ❌ | 見下方詳解。 |
| 6 | B | **B** | ✅ | 資料最小化＋去識別化＋權限控管。 |
| 7 | B | **B** | ✅ | 檢查偏差、公平性分群評估、可解釋性、申訴與人為監督。 |
| 8 | A,B,C,E | **A,B,C,E** | ✅ | 公開完整個資（D）違反治理原則。 |
| 9 | B | **C** | ❌ | 見下方詳解。 |
| 10 | B | **C** | ❌ | 見下方詳解。 |

**得分：6 / 10**

---

**第 3 題詳解**
`print` 先印 recall、後印 precision。
- 混淆矩陣：TP=2（前兩筆）、FN=2（第 3、4 筆漏抓）、FP=1（第 5 筆誤報）。
- Recall = TP/(TP+FN) = 2/4 = **0.5**
- Precision = TP/(TP+FP) = 2/3 ≈ **0.67**
- 輸出為 `0.5 0.67` → **(A)**。你選 (B) 把兩者順序看反了。

**第 5 題詳解**
對特徵尺度敏感、需縮放的是「距離型／梯度型」演算法：**KNN(A)、以梯度下降訓練的邏輯迴歸(C)、K-Means(E)**。
決策樹 (B) 與隨機森林 (D) 以「單一特徵閾值分裂」，與尺度無關，**不需縮放**。你誤選了 (D) 隨機森林、漏掉 (C) 邏輯迴歸。

**第 9 題詳解**
需求是「以條件式規則向稽核人員說明決策」＝要求**內在可解釋性**。**決策樹 (C)** 天生為 if-then 規則結構，最符合。SVM(RBF)、DNN、KNN 皆為黑箱或難以規則化。你選了 (B) SVM(RBF)，方向相反。

**第 10 題詳解**（貝氏定理／基率謬誤）
P(違規|警報) = (0.05×0.8) / (0.05×0.8 + 0.95×0.1)
= 0.04 / (0.04 + 0.095) = 0.04 / 0.135 ≈ **0.30 → (C) 約 30%**。
你選 (B) 55%，是忽略了低盛行率（5%）帶來的大量偽陽性——這正是「基率謬誤」的核心。

---

### 題型分析

錯誤集中在三類考點：

1. **評估指標的計算與定義（Q3）**：對 Recall／Precision 的公式與「輸出順序」不夠熟練。建議把混淆矩陣（TP/FP/FN/TN）畫出來再套公式，並記住 `recall_score, precision_score` 的呼叫順序即為印出順序。

2. **演算法特性的機制理解（Q5、Q9）**：
   - Q5、Q9 都答錯，反映對「各演算法底層如何運作」掌握不足。
   - 尺度敏感性取決於是否用「距離／梯度」→ 樹模型免疫。
   - 可解釋性取決於模型是否為「規則／線性」結構→ 樹模型、線性模型可解釋，SVM(RBF)、DNN、KNN 為黑箱。
   - 建議建立一張「演算法特性對照表」（是否需縮放、是否可解釋、是否適合高維、是否對離群值敏感）。

3. **機率統計／基率謬誤（Q10）**：面對「盛行率低」的診斷型題目，務必用貝氏公式而非直覺。牢記：盛行率越低，即使敏感度高，陽性預測值(PPV)仍可能偏低。

治理與倫理類（Q4、Q6、Q7、Q8）全對，觀念穩固，可保持。

---

### 新的解答建議

修正版作答：
- 第 3 題：改勾 **(A) 0.5 0.67**
- 第 5 題：改勾 **(A) KNN、(C) 邏輯迴歸(梯度下降)、(E) K-Means**（取消 D）
- 第 9 題：改勾 **(C) 決策樹**
- 其餘維持不變。

讀書建議：
1. **手算指標**：準備 3~5 組 `y_true / y_pred`，反覆手算 Accuracy／Precision／Recall／F1，並對照 sklearn 輸出驗證。
2. **演算法特性對照表**：整理縮放需求、可解釋性、對離群值/高維的敏感度，形成一頁速查。
3. **貝氏定理練習**：熟練 PPV 公式 `PPV = (盛行率×敏感度) / [盛行率×敏感度 + (1−盛行率)×偽陽率]`，多做低盛行率題型以破除基率謬誤直覺。

整體表現良好，補強「指標計算」與「演算法機制」兩塊即可再上一層。加油！💪

**Comment by JohnnyLai92** [comment]
已完成錯題分析並輸出到 `docs/ipas-ai-practice-issue-17.md`。

## Claude Code 歸檔摘要
已建立 `docs/ipas-ai-practice-issue-17.md`（未修改其他檔案）。

**本次成績：6／10**，答錯 4 題（依勾選結果比對批改留言判定）：

| 題號 | 考點 | 你的作答 | 正確答案 |
|---|---|---|---|
| 3 | L23303 評估指標 | B（0.67 0.5） | **A（0.5 0.67）**—順序看反 |
| 5 | L23301 特徵縮放 | A、D、E | **A、C、E**—誤選隨機森林、漏梯度邏輯迴歸 |
| 9 | L23202 可解釋性 | B（SVM RBF） | **C（決策樹）**—if-then 規則 |
| 10 | L23101 貝氏定理 | B（約 55%） | **C（約 30%）**—基率謬誤 |

檔案包含基本資訊、錯題列表、逐題（錯誤原因／正確觀念／相關考點）分析，以及依四大弱點排序的後續複習建議。治理與倫理類（第 4、6、7、8 題）全對，觀念穩固。
